cargo-shuttle/src/client.rs

```diff
@@ -84,16 +83,39 @@ impl ShuttleApiClient {
             .context("parsing name check response")
     }
 
-    pub async fn deploy(
+    pub async fn deploy_alpha(
         &self,
         project: &str,
         deployment_req: DeploymentRequest,
     ) -> Result<deployment::Response> {
-        let path = if self.beta {
-            format!("/projects/{project}")
+        let path = format!("/projects/{project}/services/{project}");
+        let deployment_req = rmp_serde::to_vec(&deployment_req)
+            .context("serialize DeploymentRequest as a MessagePack byte vector")?;
+
+        let url = format!("{}{}", self.api_url, path);
+        let mut builder = if self.beta {
+            self.client.put(url)
         } else {
-            format!("/projects/{project}/services/{project}")
+            self.client.post(url)
         };
+        builder = self.set_auth_bearer(builder);
```

Comment:
question: why do you still `if self.beta` when this is supposed to only be for alpha ? And there seems to be a mix between `alpha`/`beta`, because personally I would have used `alpha` for the current one, and `beta` for the new one (but better is my other suggestion to just keep the current `deploy` as-is).

Reply:
Good catch! My mistake, removed the if..else and kept only the corresponding method call (post for `deploy` and put for `deploy_beta`).